### ISSUE TITLE
DotNetCore2.0 client failing to load grains.

### ISSUE THREAD

**Issue by pms1969** [dotnet/orleans#3379]
I've tried to implement  a client in an aspnetcore project.  There are a number of test projects that require the aspnetcore testing host, and this in turn requires DependencyInjection v2.0.0

So... I've built Orleans for netcore2.0 after updating the DI version to 2.0.0 (since not doing so results in a failure to find a method), and this gets me much further, the orleans client is trying to load the grains dll, and the types in side it but returns this error:
```
ReflectionOnly loading is not supported on this platform.
Exception = System.PlatformNotSupportedException: ReflectionOnly loading is not supported on this platform.
   at System.Type.ReflectionOnlyGetType(String typeName, Boolean throwIfNotFound, Boolean ignoreCase)
   at Orleans.Runtime.CachedReflectionOnlyTypeResolver.TryPerformUncachedTypeResolution(String name, Type& type)
   at Orleans.Runtime.CachedTypeResolver.TryResolveType(String name, Type& type)
   at Orleans.Runtime.CachedTypeResolver.ResolveType(String name)
   at System.Linq.Enumerable.SelectArrayIterator`2.ToArray()
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Orleans.Runtime.AssemblyLoaderCriteria.LoadTypesAssignableFrom(Type[] requiredTypes)
   at Orleans.OutsideRuntimeClient.LoadAdditionalAssemblies()
   at Orleans.OutsideRuntimeClient.ConsumeServices(IServiceProvider services)

[2017-09-04 16:11:52.097 GMT     7	ERROR  	100319	OutsideRuntimeClient	]	!!!!!!!!!! OutsideRuntimeClient constructor failed.	
Exc level 0: System.PlatformNotSupportedException: ReflectionOnly loading is not supported on this platform.
   at System.Type.ReflectionOnlyGetType(String typeName, Boolean throwIfNotFound, Boolean ignoreCase)
   at Orleans.Runtime.CachedReflectionOnlyTypeResolver.TryPerformUncachedTypeResolution(String name, Type& type)
   at Orleans.Runtime.CachedTypeResolver.TryResolveType(String name, Type& type)
   at Orleans.Runtime.CachedTypeResolver.ResolveType(String name)
   at System.Linq.Enumerable.SelectArrayIterator`2.ToArray()
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Orleans.Runtime.AssemblyLoaderCriteria.LoadTypesAssignableFrom(Type[] requiredTypes)
   at Orleans.OutsideRuntimeClient.LoadAdditionalAssemblies()
   at Orleans.OutsideRuntimeClient.ConsumeServices(IServiceProvider services)
[2017-09-04 16:11:52.147 GMT     7	INFO   	100000	OutsideRuntimeClient	]	OutsideRuntimeClient.ConstructorReset(): client Id 	
```

The odd thing is that this doesn't happen when I kick off the silo.  It happily discovers the DLL, and loads it (although I do remember some type of error early on - but don't seem to be able to replicate this now).

Any idea's how I can get this to work?  I was thinking it might be because not all the dependencies are present in the bin folder, but from VS (which is where I'm running the tests), I can't figure out how to get all the dll's in place to allow it to pick them up?

FYI, you can see the same result, but updating the HelloWorld.NetCore app to 2.0, and ensuring the DependencyInjection version is set to 2.0.0.  I get the same result. :(

Any help appreciated.
Cheers,
Paul

**Comment by csyszf** [comment]
Same problem.

**Comment by Maarten88** [comment]
I think this one is fixed with preview3 via #3394. It now works for me!

**Comment by sergeybykov** [comment]
Resolved via #3394.
